apis/v1beta1/shared_types.go

```diff
@@ -295,6 +317,11 @@ const (
 	// match any Listeners in the Gateway.
 	RouteReasonNoMatchingParent RouteConditionReason = "NoMatchingParent"
 
+	// This reason is used with the "Accepted" condition when there are
+	// multiple ParentRefs that are not distinct. See the ParentRef
+	// definition for the definition of distinct.
+	RouteReasonParentsNotDistinct RouteConditionReason = "ParentsNotDistinct"
```

Comment:
What do we want to happen in the following case?

```yaml
parentRefs:
- name: "foo"
- name: "foo"
  sectionName: "bar"
```

Do we always want status to look something like this?

```yaml
status:
  parents:
  - parentRef:
    - name: foo
    conditions:
      type: Accepted
      status: True
  - parentRef:
    - name: foo
      sectionName: Bar
    conditions:
      type: Accepted
      status: False
      reason: ParentsNotDistinct
```

Or do we always want to consider both parents unaccepted in this state? I'd argue that it makes sense to accept one of the indistinct parents, preferably the oldest/first one.

Reply:
We talked about this one on the meeting today, and decided we should see if CEL can provide validation to protect us from these situations.